Missing dnnl_config.h when installing horovod

## Description
I tried to compile MXNet using the cmake command:
```
cmake -GNinja -C ../config.cmake ..
cmake --build .
```
and then install horovod
```
HOROVOD_WITHOUT_TENSORFLOW=1 HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_GPU_BROADCAST=NCCL pip3 install -e . --no-cache-dir --user
```
The error message is
```
/home/ubuntu/mxnet/python/mxnet/../../include/mkldnn/dnnl.hpp:23:10: fatal error: dnnl_config.h: No such file or directory
```

I find that we have these lines in `mkldnn.mk`  https://github.com/apache/incubator-mxnet/blob/c2aff58c448bbe5755de0c8d35a1cc7d3fd9aa51/mkldnn.mk#L52-L53

The problem might be that we haven't copied the files to `mxnet/include/mkldnn` so the build system used in horovod is broken.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -751,6 +751,14 @@ if(USE_DIST_KVSTORE)
   list(APPEND mxnet_LINKER_LIBS pslite)
 endif()
 
+if(USE_MKLDNN)
+    add_custom_command(TARGET mxnet POST_BUILD
+      COMMAND ${CMAKE_COMMAND} -E copy
+      ${CMAKE_BINARY_DIR}/3rdparty/mkldnn/include/dnnl_config.h  ${CMAKE_SOURCE_DIR}/include/mkldnn/
+      COMMAND ${CMAKE_COMMAND} -E copy
+      ${CMAKE_BINARY_DIR}/3rdparty/mkldnn/include/dnnl_version.h  ${CMAKE_SOURCE_DIR}/include/mkldnn/)
+endif()
+
 function(BuildTVMOP)
   # scope the variables in BuildTVM.cmake to avoid conflict
   include(cmake/BuildTVM.cmake)

